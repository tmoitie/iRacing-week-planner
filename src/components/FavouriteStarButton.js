// @flow

import * as React from 'react';
import classnames from 'classnames';

import { useTranslation } from 'react-i18next';
import styles from '../styles/main.module.scss';

type Props = {
  id: String,
  onClick: (newValue: boolean) => void,
  enabled?: boolean,
};

export default function FavouriteStarButton({ id, enabled, onClick }: Props): React.Node {
  const { t } = useTranslation();
  return (
    <span
      role="button"
      tabIndex={0}
      data-testid={id}
      onClick={() => onClick(!enabled)}
      onKeyDown={() => onClick(!enabled)}
      className={classnames({
        [styles.glyphicon]: true,
        [styles['glyphicon-star']]: enabled,
        [styles['glyphicon-star-empty']]: !enabled,
      })}
      aria-label={t(enabled ? 'Remove favorite' : 'Add favorite')}
    />
  );
}

FavouriteStarButton.defaultProps = {
  enabled: false,
};
