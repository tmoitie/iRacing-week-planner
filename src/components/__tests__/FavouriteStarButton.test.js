import React from 'react';
import { render } from '@testing-library/react';
import { describe, test } from '@jest/globals';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';

import FavouriteStarButton from '../FavouriteStarButton';

describe('components/FavouriteStarButton', () => {
  test('renders correctly', async () => {
    const user = userEvent.setup();
    const { container } = render(<FavouriteStarButton />);

    expect(container).toMatchSnapshot();
  });

  test('renders enabled with onClick', async () => {
    const user = userEvent.setup();
    const onClick = jest.fn();
    const { container } = render(<FavouriteStarButton enabled onClick={onClick} />);

    await user.click(container.firstChild);
    expect(container).toMatchSnapshot();
    expect(onClick).toHaveBeenCalledWith(false);
  });

  test('renders disabled with onClick', async () => {
    const user = userEvent.setup();
    const onClick = jest.fn();
    const { container } = render(<FavouriteStarButton onClick={onClick} />);

    await user.click(container.firstChild);
    expect(onClick).toHaveBeenCalledWith(true);
  });
});
