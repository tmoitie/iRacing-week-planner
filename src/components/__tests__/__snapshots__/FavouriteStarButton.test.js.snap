// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FavouriteStarButton renders correctly 1`] = `
<div>
  <span
    aria-label="Add favorite"
    class="glyphicon glyphicon-star-empty"
    role="button"
    tabindex="0"
  />
</div>
`;

exports[`components/FavouriteStarButton renders enabled with onClick 1`] = `
<div>
  <span
    aria-label="Remove favorite"
    class="glyphicon glyphicon-star"
    role="button"
    tabindex="0"
  />
</div>
`;
